src/coreclr/nativeaot/Runtime/unix/PalRedhawkUnix.cpp

```diff
@@ -1153,6 +1152,23 @@ REDHAWK_PALEXPORT int32_t PalGetProcessCpuCount()
     return g_RhNumberOfProcessors;
 }
 
+REDHAWK_PALEXPORT uint32_t REDHAWK_PALAPI PalGetOsPageSize()
+{
+    static int saved_pagesize = 0;
+
+    if (saved_pagesize)
+        return saved_pagesize;
+
+    // Prefer sysconf () as it's signal safe.
+#if defined (HAVE_SYSCONF) && defined (_SC_PAGESIZE)
+    saved_pagesize = sysconf(_SC_PAGESIZE);
+#else
+    saved_pagesize = getpagesize();
```

Comment:
I have notice that CoreCLR PAL uses `getpagesize`, but GC uses `sysconf(_SC_PAGESIZE)`. Is there difference between the two? Which one is better?

Reply:
the documentation seems to direct to sysconf as more portable:

```
DESCRIPTION
       The function getpagesize() returns the number of bytes in a memory page, where "page" is a fixed-length block,
       the unit for memory allocation and file mapping performed by mmap(2).

CONFORMING TO
       SVr4, 4.4BSD, SUSv2.  In SUSv2 the getpagesize() call is labeled LEGACY,  and  in  POSIX.1-2001  it  has  been
       dropped; HP-UX does not have this call.

NOTES
       Portable applications should employ sysconf(_SC_PAGESIZE) instead of getpagesize():
```

Reply:
So just call `sysconf(_SC_PAGESIZE)` everywhere without any ifdefs?

Reply:
I thought so, but mono does ifdefs and has a fallback to `getpagesize`. Perhaps the call is not always present. I noticed  that coreclr also uses sysconf conditionally. Maybe to handle some legacy cases where it is not there or for some other platforms.

My conclusion was - we can see one or another missing. If both are present, use sysconf. If neither - fail to build.

From the mono pattern it looks like it is also possible that `sysconf` is there, but `_SC_PAGESIZE` is missing.

Reply:
Is there `sysconf` on bionic or on some FreeBSD derived OS for consoles? 
I think from portability point of view mono pattern has some advantages, while not a big burden on mainstream Linux.

Reply:
GC calls sysconf unconditionally and it works fine for all platforms that we care about.

I would expect that this ifdef is only needed for some old systems that we do not care about.

Reply:
no point to be more portable than GC. I will switch to sysconf then.